api/src/main/java/me/lokka30/treasury/api/economy/transaction/EconomyTransactionInitiator.java

```diff
@@ -0,0 +1,117 @@
+package me.lokka30.treasury.api.economy.transaction;
+
+import java.util.Objects;
+import java.util.UUID;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents an initiator of {@link EconomyTransaction}.
+ * <p>Initiator is something/someone, triggering a transaction.
+ * <p>Initiators are divided in 3 types: {@link Type#PLAYER}, {@link Type#PLUGIN} and {@link Type#SERVER}.
+ * The PLAYER and PLUGIN types should always provide data which matches types:
+ * <ul>
+ *     <li>For PLAYER type: data type should be an {@link UUID} of the player, whom initiated the transaction</li>
+ *     <li>For PLUGIN type: data type should be am {@link String}, fulfilled with the plugin name, whom initiated the
+ *     transaction.</li>
+ * </ul>
+ * <p>It is not necessary to provide data for type SERVER.
```

Comment:
Since server has always the same data (no data) it could just be an instance 
```Java
  private static final EconomyTransactionInitiator<?> SERVER_INITIATOR = ...

  static EconomyTransactionInitiator<?> getServerInitiator() {
    return SERVER_INITIATOR;
  }
```
And then we could have separate static methods to create player and plugin initiators, making the comparison to see if it's the correct data easyer

Reply:
> And then we could have separate static methods to create player and plugin initiators, making the comparison to see if it's the correct data ~~easyer~~ easier

Bloat has gone too much.

Reply:
> Bloat has gone too much.

Alright. But what about the server initiator? `createInitiator` could return that instance

Reply:
We have
```java
Collections.emptyList();
```
but 
```java
new ArrayList<>();
```
also returns an empty list if not modified, doesn't it?

Reply:
I don't follow you, what are you trying to say?

Reply:
> I don't follow you, what are you trying to say?

~~learn english~~ I'm going to try to explain in code

```java
Collections.emptyList(); // isEmpty() returns true
new ArrayList<>(); // at the point of creation isEmpty() returns true
```

What does this mean? It means that you can also create an empty list even though `Collections.emptyList()` exists.

Reply:
What I'm trying to say is, why create and instance of a basically inmutable object every time is needed. Look for example at the [Optinal](https://github.com/openjdk/jdk/blob/fe2ae8e38bd8660cc637f13f425ccea17a11aa4f/src/java.base/share/classes/java/util/Optional.java) class

```Java
    private static final Optional<?> EMPTY = new Optional<>(null);

    public static <T> Optional<T> ofNullable(T value) {
        return value == null ? (Optional<T>) EMPTY
                             : new Optional<>(value);
    }

    public static<T> Optional<T> empty() {
        @SuppressWarnings("unchecked")
        Optional<T> t = (Optional<T>) EMPTY;
        return t;
    }
```
There is only one instance of an empty optional because it's basically an inmutable object, why waste time creating instances of it?

Reply:
instantiating an object in java is literally if not done in nano, then done in pico seconds. literally unmeasurable "pressure" if we count it as such. The `Optional` class is a completely different story.